src/core/Akka/Actor/ActorPath.cs

```diff
@@ -349,52 +365,208 @@ public static bool TryParse(string path, out ActorPath actorPath)
+            if (!fullScheme.StartsWith("akka"))
+                return false;
+
+            spanified = spanified.Slice(firstColonPos + 1);
+            if (!(spanified[0] == '/' && spanified[1] == '/'))
+                return false;
+
+            spanified = spanified.Slice(2);
+            var firstAtPos = spanified.IndexOf('@');
+            var sysName = string.Empty;
+
+            if (firstAtPos == -1)
+            { // dealing with an absolute local Uri
+                var nextSlash = spanified.IndexOf('/');
+
+                if (nextSlash == -1)
+                {
+                    sysName = spanified.ToString();
+                }
+                else
+                {
+                    sysName = spanified.Slice(0, nextSlash).ToString();
+                }
+
+                address = new Address(fullScheme, sysName);
+                return true;
+            }
+
+            // dealing with a remote Uri
+            sysName = spanified.Slice(0, firstAtPos).ToString();
+            spanified = spanified.Slice(firstAtPos + 1);
+
+            var secondColonPos = spanified.IndexOf(':');
+            if (secondColonPos == -1)
+                return false;
+
+            var host = spanified.Slice(0, secondColonPos).ToString();
+
+            // move past the host
+            spanified = spanified.Slice(secondColonPos + 1);
+
+            var actorPathSlash = spanified.IndexOf('/');
+            var strPort = spanified;
+            if (actorPathSlash == -1)
+            {
+                strPort = spanified;
+            }
+            else
+            {
+                strPort = spanified.Slice(0, actorPathSlash);
+            }
+
+            if (SpanHacks.TryParse(strPort, out var port))
+            {
+                address = new Address(fullScheme, sysName, host, port);
+                return true;
+            }
+
+            return false;
+
+
+            //Uri uri;
+            //return TryParseAddress(path, out address, out uri);
         }
 
-        private static bool TryParseAddress(string path, out Address address, out Uri uri)
+        private static bool TryParseAddress(string path, out Address address, out ReadOnlySpan<char> absoluteUri)
         {
-            //This code corresponds to AddressFromURIString.unapply
             address = null;
-            if (!Uri.TryCreate(path, UriKind.Absolute, out uri))
+
+            var spanified = path.AsSpan();
+            absoluteUri = spanified;
+
+            var firstColonPos = spanified.IndexOf(':');
+
+            if (firstColonPos == -1) // not an absolute Uri
                 return false;
-            var protocol = uri.Scheme; //Typically "akka"
-            if (!protocol.StartsWith("akka", StringComparison.OrdinalIgnoreCase))
-            {
-                // Protocol must start with 'akka.*
+
+            var fullScheme = spanified.Slice(0, firstColonPos).ToString().ToLowerInvariant();
```

Comment:
Stupid questions, can we use `stackalloc` here to avoid allocating two strings?

Reply:
yeah there should be a way to do the `ToLower` part without allocating a second string